de.metas.business/src/main/java/de/metas/pricing/conditions/service/impl/PricingConditionsRepository.java

```diff
@@ -291,15 +291,15 @@ private static CurrencyId extractCurrencyId(final I_M_DiscountSchemaBreak discou
 	}
 
 	@VisibleForTesting
-	/* package */ Stream<I_M_DiscountSchemaBreak> streamSchemaBreakRecords(final Properties ctx, final Collection<Integer> discountSchemaIds, final String trxName)
+	/* package */ Stream<I_M_DiscountSchemaBreak> streamSchemaBreakRecords(final Collection<Integer> discountSchemaIds)
```

Comment:
pls avoid Integer IDs.

Reply:
* Note that this was not introduced in this task. I only deleted the unnecessary ctx and trx parameters. 
* I refactored it to use PricingConditionsId .
* After refactoring there were some failing unit tests because I thought I can use the repo aware collection in the array query. It looks like we don't have it implemented yet.
=> I think it would be helpful if we implemented a version of org.adempiere.ad.dao.IQueryBuilder.addInArrayFilter(String, Collection<V>) that also works for RepoAware objects, like we have it for the Equal filter.

Reply:
Never mind. PricingConditionsId didn't implement RepoIdAware. Fixed it now.